### PR TITLE
Crs 2115 error with fixed term recall

### PR DESCRIPTION
**Pull request by joelstobart-moj**
Add handling for missing Return to Custody Date in Fixed Term Recall
- Introduced NoValidReturnToCustodyDateException to handle cases where a valid Return to Custody Date is absent in Fixed Term Recall scenarios.
- Updated `PrisonService` to throw this exception with a relevant message if decoding of Return to Custody Date fails.
- Added test coverage in `ValidationServiceTest` to verify behavior when no Return to Custody Date is present.
- Enhanced logging in `SentenceCalculationService` for better error tracing during Fixed Term Recall calculations.
- Adjusted validation logic in `RecallValidationService` to flag a missing Return to Custody Date with a specific validation message.
- Updated `CalculationIntTest` to reflect the new error message and HTTP status check for consistency.

### CODE REVIEW

**Review by mg-moj — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/RecallValidationService.kt [hunk 1]
```diff
@@ -29,6 +29,7 @@ class RecallValidationService(
     val (recallLength, has14DayFTRSentence, has28DayFTRSentence) = getFtrValidationDetails(ftrDetails, sourceData.sentenceAndOffences)
 
     return when {
+      sourceData.returnToCustodyDate == null -> listOf(ValidationMessage(ValidationCode.FTR_NO_RETURN_TO_CUSTODY_DATE))
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/RecallValidationService.kt:32]: I think you need (has14DayFTRSentence || has28DayFTRSentence) &&  sourceData.returnToCustodyDate == null

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCalculationService.kt [hunk 2]
```diff
@@ -172,9 +175,14 @@ class SentenceCalculationService(
...
   fun calculateFixedTermRecall(booking: Booking, days: Int): LocalDate {
-    return booking.returnToCustodyDate!!
-      .plusDays(days.toLong())
-      .minusDays(1)
+    try {
+      return booking.returnToCustodyDate!!
+        .plusDays(days.toLong())
+        .minusDays(1)
+    } catch (exception: Exception) {
+      log.error("There has been an error...")
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCalculationService.kt:183]: Not sure this logging is particularly helpful.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCalculationService.kt [hunk 3]
```diff
@@ -172,9 +175,14 @@ class SentenceCalculationService(
...
 
   fun calculateFixedTermRecall(booking: Booking, days: Int): LocalDate {
-    return booking.returnToCustodyDate!!
-      .plusDays(days.toLong())
-      .minusDays(1)
+    try {
+      return booking.returnToCustodyDate!!
+        .plusDays(days.toLong())
+        .minusDays(1)
+    } catch (exception: Exception) {
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCalculationService.kt:182]: Not sure catching a NPE is the best. Also how would we ever get here now its validated up front?

**Pull request merged**